src/test/java/io/pravega/connectors/flink/FlinkPravegaInputFormatTest.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2019 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.connectors.flink;
+
+
+import io.pravega.client.BatchClientFactory;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.batch.SegmentIterator;
+import io.pravega.client.batch.SegmentRange;
+import io.pravega.client.batch.StreamSegmentsIterator;
+import io.pravega.client.stream.Stream;
+import org.apache.flink.api.common.io.DefaultInputSplitAssigner;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.io.InputFormat;
+import org.junit.Test;
+
+import java.util.Iterator;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit Test for the {@link FlinkPravegaInputFormat}
+ */
+public class FlinkPravegaInputFormatTest {
+
+    private final DeserializationSchema<String> deserializationSchema = mock(DeserializationSchema.class);
+
+    private BatchClientFactory batchClientFactory = mock(BatchClientFactory.class);
+
+    private SegmentIterator<String> segmentIterator = mock(SegmentIterator.class);
+
+    private Iterator<SegmentRange> segmentRangeIterator = mock(Iterator.class);
+
+    private StreamSegmentsIterator streamSegmentsIterator = mock(StreamSegmentsIterator.class);
+
+    private final PravegaConfig pravegaConfig = mock(PravegaConfig.class);
+
+    private final Stream stream = Stream.of("test", "foo");
+
+    private final ClientConfig clientConfig = ClientConfig.builder().build();
+
+    /**
+     * Testing the builder for right configurations.
+     */
+    @Test
+    public void testBuilderForSuccess() {
+        when(pravegaConfig.getClientConfig()).thenReturn(clientConfig);
```

Comment:
These `when(...).thenReturn(...)` constructs reminded me of an issue we had in Pravega recently. Can you check whether the recommendation here is applicable?

https://github.com/mockito/mockito/issues/773

The Pravega issue for completeness is:

https://github.com/pravega/pravega/pull/3885